v2.0/inverted-indexes.md

```diff
@@ -0,0 +1,163 @@
+---
+title: Inverted Indexes
+summary: Inverted indexes improve your database's performance and usefulness by helping SQL locate schemaless data.
+toc: false
+toc_not_nested: true
+---
+
+<span class="version-tag">New in v2.0:</span> Inverted indexes improve your database's performance by helping SQL locate schemaless data (i.e., [`JSONB`](jsonb.html)).
+
+<div id="toc"></div>
+
+## How Do Inverted Indexes Work?
+
+Standard [indexes](indexes.html) work well for searches based on prefixes of sorted data. However, schemaless data like `JSONB` cannot be queried without a full table scan, since it does not adhere to ordinary value prefix comparison operators. `JSONB` needs to be indexed in a more detailed way than what a standard index provides. This is where inverted indexes prove useful.
+
+Inverted indexes filter on components of tokenizable data. The `JSONB` data type is built on two structures that can be tokenized:
+
+- **Objects** - Collections of key-value pairs where each key-value pair is a token.
+- **Arrays** - Ordered lists of values where every value in the array is a token.
+
+For example, take the following `JSONB` value in column `person`:
+
+~~~ json
+{
+  "firstName": "John",
+  "lastName": "Smith",
+  "age": 25,
+  "address": {
+    "state": "NY",
+    "postalCode": "10021"
+  }
+  "cars": [“Subaru”,
+     “Honda”]
+}
+~~~
+
+An inverted index for this object would have an entry per component, mapping it back to the original object:
+
+~~~
+"firstName": "John"
+"lastName": "Smith"
+"age": 25
+"address": "state": "NY"
+"address": "postalCode": "10021"
+"cars" : "Subaru"
+"cars" : "Honda"
+~~~
+
+This allows us to search based on subcomponents.
+
+### Creation
+
+You can use inverted indexes to improve the performance of queries using `JSONB` columns. You can create them:
+
+- At the same time as the table with the `INVERTED INDEX` clause of [`CREATE TABLE`](create-table.html#create-a-table-with-secondary-and-inverted-indexes-new-in-v2-0).
+- For existing tables with [`CREATE INVERTED INDEX`](create-index.html).
+- Using the following PostgreSQL-compatible syntax:
+
+   ~~~ sql
```

Comment:
This code block needs to be indented 4 spaces in order for it to be nested properly.

Reply:
done